gobblin-api/src/main/java/gobblin/source/extractor/Watermark.java

```diff
@@ -0,0 +1,11 @@
+package gobblin.source.extractor;
+
+import gobblin.fork.Copyable;
+
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+
+@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include=JsonTypeInfo.As.PROPERTY, property="@class")
+public interface Watermark extends Comparable<Watermark>, Copyable<Watermark> {
+
+  public void increment(Object record);
```

Comment:
I don't think I understand the point of this method. Which component is supposed to use it? This does not look like something many components should care about. The components which need to manipulate the checkpoint also should know what the implementation is. In that case, putting it in the interface is meaningless. 

Can you please make sure you add javadocs to public interfaces and methods?